test/scripts/run_test.py

```diff
@@ -176,14 +179,19 @@ def recover_deps():
 
     args = parser.parse_args()
 
+    # clean environment before test.
+    recover_deps()
+    
     if not validate_caselist(args.caselist):
         sys.exit(1)
 
     # generate sdk libraries.
-    if args.build:
-        build_libs(args.dependencies_dir)
+    if args.build and not build_libs(args.dependencies_dir):
         copy_libs()
-
+    else:
```

Comment:
So this script requires `--build-deps` in order to run test cases?

Reply:
done